plugins/usbdmx/USBDMXCom.cpp

```diff
@@ -0,0 +1,272 @@
+      frame[frame_length++] = channel & 0xff;
+      frame[frame_length++] = value;
+    }
+  }
+
+  //OLA_INFO << "CreateFrame(), frame_length=" << frame_length;
+  return frame_length;
+}
+
+/*
+ * Find the interface with the endpoint we're after. Usually this is interface
+ * 1 but we check them all just in case.
+ */
+bool LocateInterface(LibUsbAdaptor *adaptor,
+                     libusb_device *usb_device,
+                     int *interface_number) {
+  struct libusb_config_descriptor *device_config;
+  if (adaptor->GetConfigDescriptor(usb_device, 0, &device_config) != 0) {
+    OLA_WARN << "Failed to get device config descriptor";
+    return false;
+  }
+
+  OLA_DEBUG << static_cast<int>(device_config->bNumInterfaces)
+            << " interfaces found";
+  for (unsigned int i = 0; i < device_config->bNumInterfaces; i++) {
+    const struct libusb_interface *interface = &device_config->interface[i];
+    for (int j = 0; j < interface->num_altsetting; j++) {
+      const struct libusb_interface_descriptor *iface_descriptor =
+          &interface->altsetting[j];
+      for (uint8_t k = 0; k < iface_descriptor->bNumEndpoints; k++) {
+        const struct libusb_endpoint_descriptor *endpoint =
+            &iface_descriptor->endpoint[k];
+        OLA_DEBUG << "Interface " << i << ", altsetting " << j << ", endpoint "
+                  << static_cast<int>(k) << ", endpoint address "
+                  << ola::strings::ToHex(endpoint->bEndpointAddress);
+        if (endpoint->bEndpointAddress == ENDPOINT) {
+          OLA_INFO << "Using interface " << i;
+          *interface_number = i;
+          adaptor->FreeConfigDescriptor(device_config);
+          return true;
+        }
+      }
+    }
+  }
+  OLA_WARN << "Failed to locate endpoint for USBDMXCom device.";
+  adaptor->FreeConfigDescriptor(device_config);
+  return false;
+}
+}  // namespace
+
+// USBDMXComThreadedSender
+// -----------------------------------------------------------------------------
+
+/*
+ * Sends messages to a USBDMXCom device in a separate thread.
+ */
+class USBDMXComThreadedSender: public ThreadedUsbSender {
+ public:
+  USBDMXComThreadedSender(LibUsbAdaptor *adaptor,
+                          libusb_device *usb_device,
+                          libusb_device_handle *handle);
+
+ private:
+  LibUsbAdaptor* const m_adaptor;
+
+  bool TransmitBuffer(libusb_device_handle *handle,
+                      const DmxBuffer &buffer);
+};
+
+USBDMXComThreadedSender::USBDMXComThreadedSender(
+    LibUsbAdaptor *adaptor,
+    libusb_device *usb_device,
+    libusb_device_handle *usb_handle)
+    : ThreadedUsbSender(usb_device, usb_handle),
+      m_adaptor(adaptor) {
+}
+
+bool USBDMXComThreadedSender::TransmitBuffer(libusb_device_handle *handle,
+                                               const DmxBuffer &buffer) {
+  uint8_t frame[USBDMXCOM_MAX_FRAME_SIZE];
+  size_t frame_length = CreateFrame(buffer, frame);
+  if (frame_length == 0) {
+    return true;
```

Comment:
Probably chuck an OLA_INFO or DEBUG line in here so we know it deliberately hasn't sent anything.

Reply:
That would flood the console output even when nothing change, but you're the boss here

Reply:
That's why it should probably only be a DEBUG.

But at least while we're starting out it's worth having. Often people will report an issue and if you're lucky, generally all you ever get out to work on is an OLA_DEBUG level log, so the more detail the better as far as I'm concerned.

Reply:
Ok, but there is already useful ```OLA_DEBUG << "Ch. " << channel << " = " << static_cast<int>(value);``` which will be made unwatchable by the flood. It is okay to replace it by OLA_NOTICE so ?